### DISCUSSION TITLE
Unit of measure 'px' redundant

### DISCUSSION THREAD

**Discussion by umutyerebakmaz** [tailwindlabs/tailwindcss#6644] [Help] [answered]
Hello,
Tailwind adds px to the end of 0 values and I don't like this situation. In fact, as a person who has a good command of CSS, I do not write a unit of measure at the end if I am going to use the value 0 while writing css. Is there a solution for this? When I call yarn dev, tailwind produces classes like 0px. What should I do to prevent it from producing? It gets tedious to refactor every time. I think 0 is 0

```
.top-0 {
  top: 0px;
}
.right-0 {
  right: 0px;
}
.left-full {
  left: 100%;
}
.bottom-0 {
  bottom: 0px;
}
.right-full {
```

**Comment by adamwathan** [comment]
We include the unit because otherwise it is invalid when used within `calc()`. We use `calc` in Tailwind for things like `space-x-0`, so if we didn’t include the unit, that class wouldn’t work.

Originally we didn’t include the unit for zero values but it caused bugs like the one mentioned above so we added it.

**Comment by umutyerebakmaz** [comment, marked as answer]
Hello Adam, I've never looked at it from that angle. Yes you are right. Then continue to specify unit. I need to turn off these warnings in the settings of my Code Quality tool. Thank you for your answer. I support your project wholeheartedly.
